bitbots_vision/bitbots_vision/vision_modules/yoeo_handler.py

```diff
@@ -0,0 +1,207 @@
+        self._segmentations = dict()
+        self._seg_class_idx = dict()
+
+        self._load_candidate_class_names(model_path)
+
+    def _load_candidate_class_names(self, model_path: str) -> None:
+        path = join(model_path, "data", "yoeo_names.yaml")
+        data = load_classes(path)
+
+        self._det_class_names = data['detection']
+        for key, i in zip(data['segmentation'], range(len(data['segmentation']))):
+            self._seg_class_idx[key] = i
+
+    def get_candidates(self, class_name: str) -> List[Candidate]:
+        """
+        To comply with YoloHandler interface
+        """
+        return self.get_detection_candidates_for(class_name)
+
+    def get_detection_candidates_for(self, class_name: str) -> List[Candidate]:
+        assert class_name in self._det_class_names, \
+            f"Class '{class_name}' is not available for the current YOEO model (detection)"
+
+        self.predict()
+
+        return self._det_candidates[class_name]
+
+    def predict(self) -> None:
+        if self._prediction_has_to_be_updated():
+            self._compute_new_prediction()
+
+    def _prediction_has_to_be_updated(self) -> bool:
+        return self._det_candidates is None or self._segmentation is None or not self._use_caching
+
+    def get_classes(self) -> List[str]:
+        """
+        To comply with YoloHandler interface
+        """
+        return self.get_detection_candidate_class_names()
+
+    def get_detection_candidate_class_names(self) -> List[str]:
+        return self._det_class_names
+
+    def get_segmentation_candidate_class_names(self):
+        return self._seg_class_idx
+
+    def get_segmentation_for(self, class_name: str):
+        assert class_name in self._seg_class_idx, \
+            f"Class '{class_name}' ist not available for the current YOEO model (segmentation)"
+
+        self.predict()
+
+        if class_name not in self._segmentations.keys():
+            self._compute_segmentation_for(class_name)
+
+        return self._segmentations[class_name]
+
+    def _compute_segmentation_for(self, class_name: str) -> None:
+        segmentation = where(self._segmentation == self._seg_class_idx[class_name], 1, 0)
+        self._segmentations[class_name] = moveaxis(segmentation, 0, -1)
+
+    def set_image(self, img) -> None:
+        self._reset()
+        self._update_image(img)
+
+    def _reset(self) -> None:
+        self._det_candidates = defaultdict(list)
+        self._segmentation = None
+        self._segmentations = defaultdict(list)
+
+    def _update_image(self, img) -> None:
+        self._image = img
+
+    @abstractmethod
+    def _compute_new_prediction(self) -> None:
+        ...
+
+
+class YOEOHandlerPytorch(YOEOHandlerTemplate):
+    """
+    Using Pytorch to get YOEO predictions
+    """
+
+    def __init__(self, config, model_path):
+        super().__init__(config, model_path)
+
+        weights_path = join(model_path, "weights", "yoeo.pth")
```

Comment:
See model import for the vision described above.

Reply:
?

Reply:
I wanted to say that the model seems to be loaded from the YOEO repo and not the models folder inside the vision which is downloaded from http://data.bit-bots.de/models/.

Reply:
You can just do 

```python
weights_path = join(model_path, "yoeo.pth")
```

if I recall correctly.

Reply:
Ahh, ok. I'll change that.

Reply:
And what about the cfg file?

Reply:
It is in the same folder and called `yoeo.cfg`